test/ready/shared.dart

```diff
@@ -2,97 +2,60 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'package:scheduled_test/scheduled_test.dart';
+import 'package:test/test.dart';
 
 import '../utils.dart';
 
 void sharedTests() {
-  test('ready does not complete until after subscription', () {
-    var watcher = createWatcher(waitForReady: false);
-
-    var ready = false;
-    watcher.ready.then((_) {
-      ready = true;
-    });
```

Comment:
It's important to check when the future fired—we're testing that it fires at the correct time, and we shouldn't rely on it matching `watcher.isReady`. Same below.

Reply:
You still need to test *when* `watcher.ready` completes—it shouldn't be complete before `watcher.events` has a listener, and it should be complete after `watcher.ready` fires.

Reply:
Should be good now.